native-sql-engine/core/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala

```diff
@@ -548,30 +549,26 @@ case class ColumnarOverrideRules(session: SparkSession) extends ColumnarRule wit
   def preOverrides = ColumnarPreOverrides(session)
   def postOverrides = ColumnarPostOverrides()
 
-  def columnarWholeStageEnabled = conf.getBoolean("spark.oap.sql.columnar.wholestagecodegen", defaultValue = true) && !codegendisable
+  def columnarWholeStageEnabled = conf.getBoolean(
+    "spark.oap.sql.columnar.wholestagecodegen", defaultValue = true) && !codegendisable
   def collapseOverrides = ColumnarCollapseCodegenStages(columnarWholeStageEnabled)
+  def enableArrowColumnarToRow: Boolean =
+    conf.getBoolean("spark.oap.sql.columnar.columnartorow", defaultValue = true)
+  def wholeStageFallbackThreshold: Int =
+    conf.getInt("spark.oap.sql.columnar.wholeStage.fallback.threshold", defaultValue = 4)
 
   var isSupportAdaptive: Boolean = true
 
-  private def supportAdaptive(plan: SparkPlan): Boolean = {
-    // TODO migrate dynamic-partition-pruning onto adaptive execution.
-    // Only QueryStage will have Exchange as Leaf Plan
-    val isLeafPlanExchange = plan match {
-      case e: Exchange => true
-      case other => false
-    }
-    isLeafPlanExchange || (SQLConf.get.adaptiveExecutionEnabled && (sanityCheck(plan) &&
-    !plan.logicalLink.exists(_.isStreaming) &&
-    !plan.expressions.exists(_.find(_.isInstanceOf[DynamicPruningSubquery]).isDefined) &&
```

Comment:
This part of code is ported from spark's source code. From spark 3.2, DPP can be coexisted with AQE, so the below statement should be removed for spark 3.2.  We differentiated the implementation in shim layers.
```
!plan.expressions.exists(_.find(_.isInstanceOf[DynamicPruningSubquery]).isDefined
```